In [ ]:
import pandas as pd
import numpy as np

In [ ]:
data={
    "Name":["Asha","Ravi","Priya","Kiran","Rahul"],
    "Age":[20,21,20,22,21],
    "Marks":[85,78,92,65,88],
    "Department":["CSE","ECE","CSE","EEE","CSE"]
}
df=pd.DataFrame(data)
print(df)

    Name  Age  Marks Department
0   Asha   20     85        CSE
1   Ravi   21     78        ECE
2  Priya   20     92        CSE
3  Kiran   22     65        EEE
4  Rahul   21     88        CSE


In [ ]:
print(df.head())

    Name  Age  Marks Department
0   Asha   20     85        CSE
1   Ravi   21     78        ECE
2  Priya   20     92        CSE
3  Kiran   22     65        EEE
4  Rahul   21     88        CSE


In [ ]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Name        5 non-null      object
 1   Age         5 non-null      int64 
 2   Marks       5 non-null      int64 
 3   Department  5 non-null      object
dtypes: int64(2), object(2)
memory usage: 292.0+ bytes
None


In [ ]:
print(df.describe())

            Age      Marks
count   5.00000   5.000000
mean   20.80000  81.600000
std     0.83666  10.597169
min    20.00000  65.000000
25%    20.00000  78.000000
50%    21.00000  85.000000
75%    21.00000  88.000000
max    22.00000  92.000000


In [ ]:
print(df.isnull().sum())

Name          0
Age           0
Marks         0
Department    0
dtype: int64


In [ ]:
df=df.drop_duplicates()
print(df)

    Name  Age  Marks Department
0   Asha   20     85        CSE
1   Ravi   21     78        ECE
2  Priya   20     92        CSE
3  Kiran   22     65        EEE
4  Rahul   21     88        CSE


In [ ]:
high_scores=df[df["Marks"]>80]
print(high_scores)

    Name  Age  Marks Department
0   Asha   20     85        CSE
2  Priya   20     92        CSE
4  Rahul   21     88        CSE


In [ ]:
sorted_data=df.sort_values(by="Marks",ascending=False)
print(sorted_data)

    Name  Age  Marks Department
2  Priya   20     92        CSE
4  Rahul   21     88        CSE
0   Asha   20     85        CSE
1   Ravi   21     78        ECE
3  Kiran   22     65        EEE


In [ ]:
print("Average Marks:",df["Marks"].mean())
print("Highest Marks:",df["Marks"].max())
print("Lowest Marks:",df["Marks"].min())

Average Marks: 81.6
Highest Marks: 92
Lowest Marks: 65


In [ ]:
print(df.groupby("Department")["Marks"].mean())

Department
CSE    88.333333
ECE    78.000000
EEE    65.000000
Name: Marks, dtype: float64


Conclusion

The student dataset was successfully explored and analyzed using Python, Pandas, and NumPy.

The analysis included data inspection, missing-value checking, duplicate removal, filtering, sorting, grouping, and statistical calculations.

The average student mark is 81.6, while the highest mark is 92 and the lowest mark is 65.